src/envs/painting.py

```diff
@@ -0,0 +1,561 @@
+"""Painting domain, which allows for two different grasps on an object
+(side or top). Side grasping allows for placing into the shelf, and top
+grasping allows for placing into the box. The box has a lid which
+may need to be opened; this lid is NOT modeled by any of the given
+predicates, but could be modeled by a learned predicate.
```

Comment:
but we know not to expect this anytime soon because of the action at a distance issue, so maybe just remove the latter clause